lib/ansible/modules/packaging/os/zypper_repository.py

```diff
@@ -389,23 +440,29 @@ def exit_unchanged():
     else:
         shortname = alias
 
-    if state == 'present':
+    if action == 'add' and state == 'present':
         if exists and not mod:
             if runrefresh:
                 runrefreshrepo(module, auto_import_keys, shortname)
             exit_unchanged()
         rc, stdout, stderr = addmodify_repo(module, repodata, old_repos, zypper_version, warnings)
         if rc == 0 and (runrefresh or auto_import_keys):
             runrefreshrepo(module, auto_import_keys, shortname)
-    elif state == 'absent':
+    elif action == 'modify' and state == 'present':
+        if exists and not mod:
+            exit_unchanged()
+        if not exists:
+            exit_unchanged()
+        rc, stdout, stderr = modify_repo(module, repodata, zypper_version, warnings)
+    elif state == 'absent' or action == 'remove':
         if not exists:
             exit_unchanged()
         rc, stdout, stderr = remove_repo(module, shortname)
 
     if rc == 0:
-        module.exit_json(changed=True, repodata=repodata, state=state, warnings=warnings)
+        module.exit_json(changed=True, repodata=repodata, action=action, state=state, warnings=warnings)
     else:
-        module.fail_json(msg="Zypper failed with rc %s" % rc, rc=rc, stdout=stdout, stderr=stderr, repodata=repodata, state=state, warnings=warnings)
+        module.fail_json(msg="zypper failed with rc %s" % rc, rc=rc, stdout=stdout, stderr=stderr, repodata=repodata, action=action, state=state, warnings=warnings)
```

Comment:
ok, I am very new to python ... how can I break this obviously too long line into two lines ?
Can you help please ?
Thank you